### PR TITLE
Update: disable self-update for managed deployments

### PR DESCRIPTION
**Pull request by h4x3rotab**
## Summary

- Problem: Clawdi Phala deployments can still trigger OpenClaw's in-place self-update paths, including package-manager installs that replace the managed Clawdi distribution with incompatible vanilla OpenClaw bits.
- Why it matters: a user-triggered self-update can brick the deployment, especially because the rendered Clawdi/Phala config is not compatible with the vanilla upstream package/update flow.
- What changed: added `update.selfUpdate.enabled/reason`, block package-install self-update when disabled, and set the Clawdi Phala template to disable self-update with an operator-facing reason.
- What did NOT change (scope boundary): this does not hide update UI/actions, change git-checkout update behavior, or add a Phala migration script.

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Docs
- [x] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [x] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [x] API / contracts
- [x] UI / DX
- [x] CI/CD / infra

## Linked Issue/PR

- Related https://github.com/Clawdi-AI/clawdi/issues/174

## User-visible / Behavior Changes

- Managed deployments can set `update.selfUpdate.enabled=false` and an optional `update.selfUpdate.reason`.
- When disabled, package-manager self-update paths now return `skipped` with the configured reason instead of running `npm`/`pnpm`/`bun` installs.
- The Clawdi Phala template now ships with self-update disabled by default.

## Security Impact (required)

- New permissions/capabilities? (`No`)
- Secrets/tokens handling changed? (`No`)
- New/changed network calls? (`No`)
- Command/tool execution surface changed? (`Yes`)
- Data access scope changed? (`No`)
- If any `Yes`, explain risk + mitigation:
  - This narrows the update execution surface for managed deployments by refusing in-place package installs when the config disables self-update.

### Environment

- OS: Ubuntu (dev workspace)
- Runtime/container: Node 24 / pnpm
- Model/provider: n/a
- Integration/channel (if any): n/a
- Relevant config (redacted): `update.selfUpdate.enabled=false`

### Steps

1. Configure a package-installed OpenClaw deployment with `update.selfUpdate.enabled=false`.
2. Run `openclaw update` or trigger `update.run` through the gateway/control plane.
3. Observe the result.

### Expected

- The update is skipped with a clear operator-facing reason.
- No package-manager install command is executed.

### Actual

- Before this change, package-manager self-update could run and replace the managed Clawdi install.

## Evidence

- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

## Human Verification (required)

What you personally verified (not just CI), and how:

- Verified scenarios:
  - `src/infra/update-runner.test.ts` covers package-manager update blocking.
  - `src/cli/update-cli.test.ts` covers both normal package updates and package-to-dev switch flows being blocked when self-update is disabled.
  - `pnpm build` passed.
- Edge cases checked:
  - package install path
  - package-to-dev/git-switch path
  - explicit disabled reason propagation
- What you did **not** verify:
  - a live Phala deployment after config rollout

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? (`Yes`)
- Config/env changes? (`Yes`)
- Migration needed? (`No`)
- If yes, exact upgrade steps:

## Failure Recovery (if this breaks)

- How to disable/revert this change quickly:
  - Remove `update.selfUpdate` from config or set `update.selfUpdate.enabled=true`.
- Files/config to restore:
  - `openclaw.json`
  - `phala-deploy/openclaw.template.json`
- Known bad symptoms reviewers should watch for:
  - package-installed non-managed environments unexpectedly seeing update skips because `update.selfUpdate.enabled=false` was set in config

## Risks and Mitigations

- Risk:
  - Managed-deployment opt-out could accidentally be applied to an environment that still expects in-place package updates.
- Mitigation:
  - The block is fully config-driven, returns a clear reason, and defaults to unchanged behavior unless the flag is explicitly set.

### CODE REVIEW

File: phala-deploy/openclaw.template.json [hunk 1]
```diff
@@ -39,7 +39,11 @@
     }
   },
   "update": {
-    "checkOnStart": false
+    "checkOnStart": false,
+    "selfUpdate": {
+      "enabled": false,
+      "reason": "This Clawdi deployment is managed by the Phala image rollout workflow. In-place OpenClaw self-update would replace it with an incompatible upstream package and can brick the deployment."
```
Comment (h4x3rotab) [inline phala-deploy/openclaw.template.json]: Mention "please use clawdi dashboard to update", and make it slightly less technical

Comment (h4x3rotab) [reply phala-deploy/openclaw.template.json]: Updated in 4eeb684dfe. The template now tells users to use the Clawdi dashboard to update this deployment, without the more technical rollout details.
